### PR TITLE
Initial version of helper.ts for CW721-base

### PR DESCRIPTION
**Pull request by volkrass**
This is my initial version of helper.ts for CW721-base contract. 

npx @cosmjs/cli@^0.23 --init contracts/cosmons/helpers.ts 

Following methods are implemented and will be part of helper.ts - comments below will be part of next version. Here for review:

### Setup Contract
```
const client = await useOptions(defaultOptions).setup(<YOUR PASSWORD>);
const partner = await useOptions(defaultOptions).setup(<YOUR PASSWORD>, ".wallet2.key");
const cw721 = CW721(client);
const codeId = <your CodeID>;
const initMsg = { name: "Cosmons", symbol: "mons",  minter: address };
const contract = await client.instantiate(codeId, initMsg, "Virtual Cosmons 1");
const mine = cw721.use(contract.contractAddress);
```

### Transfer Token to Partner
`mine.transferNft(partnerAddr, "monster112a9lf95atqvyejqe22xnna8x4mfqd75tkq2kvwcjyysarcsb");
`
### Approve Contract
// Don't call transferNft before, otherwise it fails
`mine.approve(address, "monster112a9lf95atqvyejqe22xnna8x4mfqd75tkq2kvwcjyysarcsb");
`
### Queries

```
mine.nftInfo("monster112a9lf95atqvyejqe22xnna8x4mfqd75tkq2kvwcjyysarcsb")
mine.ownerOf("monster112a9lf95atqvyejqe22xnna8x4mfqd75tkq2kvwcjyysarcsb")
mine.numTokens()
mine.tokens(address, "", 10)
mine.allNftInfo("monster112a9lf95atqvyejqe22xnna8x4mfqd75tkq2kvwcjyysarcsb")
mine.allTokens("", 10)
```

### CODE REVIEW

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:50]: These two are pointing to the same network with different faucet, right? Why not unify them?

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:96]: I think with 0.23 this becomes a lot simpler. But good to stick with what is working for now, we (Confio) can upgrade this to 0.23 after it is merged (and you wish to upgrade)

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:254]: start_after and limit are optional here as well

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:248]: this should be `tokenId: string` not `owner: string`

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [reply contracts/cw721-base/helpers.ts:254]: Actually, probably nicest to use `startAfter` for the interface and convert in the json sent over the wire.

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:258]: Also takes `image?: string` I believe

Comment (ethanfrey) [reply contracts/cw721-base/helpers.ts:258]: I think defining `type TokenId = string` somewhere above and using it some places would make the return values clearer - no runtime or type-check difference, just more readable. Actually, what is the string returned here... I guess not the TokenId.

Comment (ethanfrey) [reply contracts/cw721-base/helpers.ts:258]: Oh, and for public API, nice to use `tokenId` rather than `token_id`

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:295]: :+1:

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:331]: this is defined as `description?: string` in the interface above. good to be consistent. (and nice to add image url here as well)

File: contracts/cw721-base/helpers.ts
Comment (ethanfrey) [inline contracts/cw721-base/helpers.ts:337]: Ah, they all return the transactionHash. That is great.
Maybe you can write a type alias `type TransactionHash = string` and use it in the interface to make it clear to someone calling it what the string is.

**Review by ethanfrey — COMMENTED** [review]
Nice work here.

Added a lot of very minor comments on it, just some tips to polish it up to a very high quality interface.

Hope this is working well for you in your UI.

**Review by ethanfrey — COMMENTED** [review]
Nice cleanup!

**Comment by BlockscapeLab** [comment]
@ethanfrey, from my point of view helper.ts is ready to merge. Let me know if there is something what I should improve.

**Comment by ethanfrey** [comment]
Sorry for the delay reviewing this. I have been on holiday.

I will check this now. But only one of the 2 github committers has signed the CLA. Could you fix this?

**Review by ethanfrey — COMMENTED** [review]
Looks quite good. Some small changes requested.

Mainly just replacing any with concrete types.

**Comment by volkrass** [comment]
@ethanfrey , I guess we can close this ticket. My collegues has been fixed the issues, so that this pull request is outdated. What do you think?

**Review by ethanfrey — APPROVED** [review]
Sorry for the delay I missed the notification that this was fixed and it slipped.

Merging now,

**Pull request merged**
